beep/protocol_tools/arbin_schedule_file.py

```diff
@@ -0,0 +1,251 @@
+                variables.
+        """
+        values = []
+        s = '[Schedule]'
+        for sch_keys in sdu_dict[s].keys():
+            if re.search(self.step, sch_keys) and sdu_dict[s][sch_keys]['m_szLabel'] == step_label:
+                if step_value is not None:
+                    sdu_dict[s][sch_keys][step_key] = step_value
+                values.append(sdu_dict[s][sch_keys][step_key])
+        return sdu_dict
+
+    def step_limit_values(self, sdu_dict, step_label, limit_var, limit_set=None):
+        """Insert values for the limits in the steps in the schedule section
+
+        Args:
+            sdu_dict (dict): Ordered dictionary containing all of the schedule file
+            step_label (str): The user determined step label for the step. If there
+                are multiple identical labels this will operate on the first one it
+                encounters
+            limit_var (str): Variable being used for this particular limit in the step
+            limit_set (dict): Value comparison to trip the limit (evaluating to True
+                triggers the limit) {'compare': '>', 'value': '0.086'}
+
+        Returns:
+            dict: Altered ordered dictionary with keys corresponding to options or control
+                variables.
+        """
+        values = []
+        s = '[Schedule]'
+        equ = 'Equation0_sz'
+        for sch_keys in sdu_dict[s].keys():
+            if re.search(self.step, sch_keys) and sdu_dict[s][sch_keys]['m_szLabel'] == step_label:
+                for step_keys in sdu_dict[s][sch_keys].keys():
+                    if re.search(self.limit, step_keys) and \
+                            sdu_dict[s][sch_keys][step_keys][equ + 'Left'] == limit_var and \
+                            sdu_dict[s][sch_keys][step_keys]['m_bStepLimit'] == '1':
+                        if limit_set is not None:
+                            sdu_dict[s][sch_keys][step_keys][equ + 'CompareSign'] = limit_set['compare']
+                            sdu_dict[s][sch_keys][step_keys][equ + 'Right'] = limit_set['value']
+                        values.append(sdu_dict[s][sch_keys][step_keys][equ + 'Left'] +
+                                      sdu_dict[s][sch_keys][step_keys][equ + 'CompareSign'] +
+                                      sdu_dict[s][sch_keys][step_keys][equ + 'Right'])
+                    elif re.search(self.limit, step_keys) and \
+                            sdu_dict[s][sch_keys][step_keys]['m_bStepLimit'] == '1':
+                        print('Warning, additional step limit: ' +
+                              sdu_dict[s][sch_keys][step_keys][equ + 'Left'] +
+                              sdu_dict[s][sch_keys][step_keys][equ + 'CompareSign'] +
+                              sdu_dict[s][sch_keys][step_keys][equ + 'Right'])
+        return sdu_dict
+
+
+def main():
+    sdu = ScheduleFile(version='0.1')
+    sdu.fast_charge_file(1.1*3.6, 0.086, 1.1*5, '20170630-3_6C_9per_5C.sdu', 'test.sdu')
```

Comment:
Are the above CC1 and CC2 values optimal ones from the fast charge work?

Reply:
No, they just happen to be the values in the schedule file that I had. I think this is actually a schedule file from the FastCharge data set